wire-runtime/src/main/java/com/squareup/wire/FieldBinding.java

```diff
@@ -48,27 +49,49 @@ private static Method getBuilderMethod(Class<?> builderType, String name, Class<
   public final int tag;
   public final ProtoType type;
   public final boolean redacted;
-  public final ProtoAdapter<?> singleAdapter;
-  public final ProtoAdapter<?> adapter;
-
   private final Field messageField;
   private final Field builderField;
   private final Method builderMethod;
 
-  FieldBinding(WireField wireField, ProtoAdapter<?> singleAdapter,
-      Field messageField, Class<B> builderType) {
+  // Delegate adapters are created lazily; otherwise we could stack overflow!
```

Comment:
(no more deferred adapter. this laziness goes away when we have static constants!)